### PR TITLE
feat: move non-subcircuit group

### LINKED ISSUE #390: Non-subcircuit Groups don't properly offset schematic components, can cause traces not to render
https://github.com/user-attachments/assets/73f24fe9-dde2-45cf-a0c9-0ba99934cb29

/bounty $40

### PR DESCRIPTION
**Pull request by imrishabh18**
/close #390

### CODE REVIEW

**Review by seveibar — COMMENTED** [review]
Not a bad implementation but i have a slightly different one in mind that i think coupd be better- each parent returns it’s “transform” to its children (something like _computeSchematicTransform), this is better than offsetting in a later render phase.

I can give it a try shortly, feel free to take a stab!

**Comment by seveibar** [comment]
Basically _computeSchematicPositionBeforeLayout should return the correct schematic position for children (offset by all parent groups)

**Comment by imrishabh18** [comment]
@seveibar This is what you mean't by having another function to compute the offset for the decedent's?

File: lib/components/primitive-components/Group/Group.ts [hunk 1]
```diff
@@ -308,18 +308,48 @@ export class Group<Props extends z.ZodType<any, any, any> = typeof groupProps>
     }
   }
 
+  _computeSchematicPositionBeforeLayout(): void {
+    const { db } = this.root!
```
Comment (seveibar) [inline lib/components/primitive-components/Group/Group.ts:312]: This method shouldnt mutate, never mutate db unless you're definitely inside a render phase and the name of the method makes sense to mutate

**Review by seveibar — CHANGES_REQUESTED** [review]

**Comment by seveibar** [comment]
@imrishabh18 i'll record a video for you

**Pull request closed without merging**